src/FakeItEasy/Core/DefaultFakeManagerAccessor.cs

```diff
@@ -26,7 +26,21 @@ public FakeManager GetFakeManager(object proxy)
 
             if (result == null)
             {
-                throw new ArgumentException("The specified object is not recognized as a fake object.");
+                object tag = null;
+                ITaggable taggable = proxy as ITaggable;
+                if (taggable != null && taggable.Tag != null)
+                {
+                    tag = taggable.Tag;
+                }
+
+                if (tag == null)
+                {
+                    throw new ArgumentException(string.Format(ExceptionMessages.NotRecognisedAsFake, proxy.GetType().FullName));

```

Comment:
This is not something I'd expect a first-time contributor to know, but we describe types using an extension method, `TypeExtensions.FullNameCSharp`. If you know to look, you can see it all throughout the code, and no uses of `FullName` (for description) in the production code. Would you change to use the extension method, to provide consistent rendering?